core-library/src/main/java/org/silverpeas/core/index/search/model/IndexSearcher.java

```diff
@@ -150,27 +146,27 @@ public static int getFactorFromProperties(String propertyName, int defaultValue)
    * @param objectType
    * @return MatchingIndexEntry wrapping the result, else null
    */
-  public MatchingIndexEntry search(String component, String objectId, String objectType) {
+  public MatchingIndexEntry search(String component, String objectId, String objectType)
+      throws IOException {
     Set<String> set = new HashSet<>(1);
     set.add(component);
 
     IndexEntryKey indexEntryKey = new IndexEntryKey(component, objectType, objectId);
     MatchingIndexEntry matchingIndexEntry = null;
     org.apache.lucene.search.IndexSearcher searcher = getSearcher(set);
     try {
-      TopDocs topDocs;
       Term term = new Term(IndexManager.KEY, indexEntryKey.toString());
 
       TermQuery query = new TermQuery(term);
-      topDocs = searcher.search(query, maxNumberResult);
+      TopDocs topDocs = searcher.search(query, maxNumberResult);
       ScoreDoc scoreDoc = topDocs.scoreDocs[0];
 
       matchingIndexEntry = createMatchingIndexEntry(scoreDoc, "*", searcher);
     } catch (IOException ioe) {
       SilverLogger.getLogger(this).error("Index file corrupted", ioe);
-    } finally {
+    } /*finally {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this trailing comment on the previous empty line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3ATrailingCommentCheck)